src/sentry/api/endpoints/project_transaction_threshold.py

```diff
@@ -89,9 +89,10 @@ def post(self, request, project):
                 project_threshold.edited_by = request.user
                 project_threshold.save()
 
-                created = False
+            created = False
 
-            except ProjectTransactionThreshold.DoesNotExist:
+        except ProjectTransactionThreshold.DoesNotExist:
+            with transaction.atomic():
```

Comment:
Linking a comment that @Zylphrex  had from a different PR pertaining to these changes
https://github.com/getsentry/sentry/pull/26824/files#r660760626

>Is it possible to hit an integrity error here if the threshold gets created between the get from above and the create from below?

Reply:
Don't have a solution for this atm